server/channels/app/platform/web_conn.go

```diff
@@ -843,7 +883,17 @@ func (wc *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 	}
 
 	// Only report events to users who are in the channel for the event
-	if msg.GetBroadcast().ChannelId != "" {
+	if chID := msg.GetBroadcast().ChannelId; chID != "" {
+		// For typing events, we don't send them to users who don't have
+		// that channel or thread opened.
+		if wc.Platform.Config().FeatureFlags.WebSocketEventScope &&
```

Comment:
I am being a bit pedantic here but if the feature flag is used only here it's purely about preventing specific server side behaviour vs protecting against bugs/regressions which I think it's generally what we want to do. That's why I was a bit worried of polluting the code with checks.

Reply:
Just being a bit practical here. The only real change is not sending the event. If we can control that, we can control the feature. Were you thinking of adding the flag on client side too?

Reply:
Yeah, that would be my expectation from a feature flag, to cover the whole feature.